lib/cfa/grub2/device_map.rb

```diff
@@ -0,0 +1,71 @@
+require "cfa/base_model"
+require "cfa/augeas_parser"
+require "cfa/placer"
+require "cfa/matcher"
+
+module CFA
+  module Grub2
+    # Represents grub device map in /etc/grub2/device_map
+    # for details see https://www.gnu.org/software/grub/manual/html_node/Device-map.html
+    # Main features:
+    #
+    # - Do not overwrite files
+    # - When setting value first try to just change value if key already exists
+    # - When grub key is not there, then add to file
+    # - checks and raise exception if number of mappings exceed limit 8.
+    #   Limitation is caused by BIOS Int 13 used by grub2 for selecting boot
+    #   device.
+    class DeviceMap < BaseModel
+      PARSER = AugeasParser.new("device_map.lns")
+      PATH = "/etc/grub2/device.map"
+
+      def initialize(file_handler: File)
+        super(PARSER, PATH, file_handler: file_handler)
+        # TODO: add to parser method to fill empty data tree
+        self.data = AugeasTree.new
+      end
+
+      def save(changes_only: false)
+        raise "Too much grub devices. Limit is 8." if grub_devices.size > 8
```

Comment:
Wouldn't it be better to have a custom exception class instead of raising `RuntimeError`?